packages/storage/amplify_storage_s3/lib/src/S3PrefixResolver/prefix_handler.dart

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_core/amplify_core.dart';
+
+import '../types.dart';
+
+class PrefixHandler {
+  static Future<dynamic> handlePrefix(Map<String, dynamic> arguments,
+      AmplifyStorageS3PrefixResolver prefixResolver) async {
+    final accessLevelString = arguments['accessLevel'];
+    StorageAccessLevel accessLevel;
+    switch (accessLevelString) {
+      case 'public':
+        // Why are our storageAccessLevel enums different?
+        accessLevel = StorageAccessLevel.guest;
+        break;
+      case 'guest':
+        accessLevel = StorageAccessLevel.guest;
+        break;
+      case 'protected':
+        accessLevel = StorageAccessLevel.protected;
+        break;
+      case 'private':
+        accessLevel = StorageAccessLevel.private;
+        break;
+      default:
+        throw StateError("Native sent invalid accessLevelString");
+    }
+    String? targetIdentity = arguments['targetIdentity'];
+
+    try {
+      String prefix = await prefixResolver.resolvePrefix(
+          storageAccessLevel: accessLevel, identityId: targetIdentity);
+      return {'isSuccess': true, 'prefix': prefix};
+      // Native catches StorageExceptions, but why would customer code throw an Amplify exception?
+    } on AmplifyException catch (e) {
```

Comment:
What about other exception types? If the user is implementing `resolvePrefix`, they may not (probably won't) be throwing AmplifyExceptions

Reply:
Updated to: 

```
on Exception catch (e) {
      return {
        'isSuccess': false,
        'errorMessage': e.toString(),
        'errorRecoverySuggestion': ''
      };
```

Reply:
I think we can use the [`StorageException`](https://github.com/aws-amplify/amplify-flutter/blob/main/packages/amplify_core/lib/src/types/storage/exceptions/storage_exception.dart) here.

Reply:
Switched to catch generic Exception, see updated code

Reply:
@HuiSF would customer code throw an amplify StorageException though?

Reply:
The generic exception type makes sense 👍